src/runtime_src/tools/xclbinutil/Section.cxx

```diff
@@ -31,15 +32,81 @@ namespace XUtil = XclBinUtilities;
 #endif
 
 
+Section::SectionInfo::SectionInfo( enum axlf_section_kind eKind,
+                                   const std::string & sectionName,
+                                   Section_factory sectionCtor)
+  : m_eKind(eKind)
+  , m_sectionName(sectionName)
+  , m_sectionCtor(sectionCtor)
+  , m_nodeName("")
+  , m_bSupportsSubSections(false)
+  , m_bSupportsIndexing(false)
+{
+  // Empty
+}
+
+void 
+Section::SectionInfo::setNodeName(const std::string& sNodeName)
+{
+  m_nodeName = sNodeName;
+}
+
+void 
+Section::SectionInfo::setSupportsSubSections(bool bValue)
+{
+  m_bSupportsSubSections = bValue;
+}
 
-// Static Variables Initialization
-std::map<enum axlf_section_kind, std::string> Section::m_mapIdToName;
-std::map<std::string, enum axlf_section_kind> Section::m_mapNameToId;
-std::map<enum axlf_section_kind, Section::Section_factory> Section::m_mapIdToCtor;
-std::map<std::string, enum axlf_section_kind> Section::m_mapJSONNameToKind;
-std::map<enum axlf_section_kind, std::string> Section::m_mapKindToJSONName;
-std::map<enum axlf_section_kind, bool> Section::m_mapIdToSubSectionSupport;
-std::map<enum axlf_section_kind, bool> Section::m_mapIdToSectionIndexSupport;
+void 
+Section::SectionInfo::setSupportsIndexing(bool bValue)
+{
+  m_bSupportsIndexing = bValue;
+}
+
+enum axlf_section_kind 
+  Section::SectionInfo::getType() const
+{
+  return m_eKind;
+}
+
+const std::string &
+Section::SectionInfo::getSectionName() const
+{
+  return m_sectionName;
+}
+
+Section::Section_factory
+Section::SectionInfo::getCtor() const
+{
+  return m_sectionCtor;
+}
+
+
+const std::string &
+Section::SectionInfo::getNodeName() const
+{
+  return m_nodeName;
+}
```

Comment:
Being that the SectionInfo class is meant to represent a collection of data, in many ways it can be considered a structure.  If it was we would be accessing the variables directly.  In other words, I have no issues with exposing these variables.

Note
Setters and getters have always bother me to, especially with primatives like this.  I will remove them and access the variables directly.  If anything, the code will be smaller and more simple.